### PR TITLE
Set CMAKE_CXX_STANDARD so cmake does not duplicate std setting

### PR DESCRIPTION
**Pull request by jacobschloss**
The current CMake scripts cause 3 different language standards to be set when building. One by cmake/EnableC++XX.cmake, one by gazebo, and a default one that CMake will append.

Neither the language standards that EnableC++XX.cmake set nor that were in GAZEBO_CXX_FLAGS were actually being used, since cmake's was last on the arguments.

Before this patch c++17, c++11, and gnu++14 is set at the same time.

> /usr/bin/c++  -DLIBBULLET_VERSION=2.87 -DLIBBULLET_VERSION_GT_282 -Dmav_msgs_EXPORTS -isystem /usr/include/gazebo-9 -isystem /usr/include/bullet -isystem /usr/include/simbody -isystem /usr/include/sdformat-6.0 -isystem /usr/include/ignition/math4 -isystem /usr/include/OGRE -isystem /usr/include/OGRE/Terrain -isystem /usr/include/OGRE/Paging -isystem /usr/include/ignition/transport4 -isystem /usr/include/ignition/msgs1 -isystem /usr/include/ignition/common1 -isystem /usr/include/ignition/fuel_tools1 -I/home/jschloss/firmware/Tools/sitl_gazebo/include -I/home/jschloss/firmware/build/posix_sitl_default/build_gazebo -I/usr/include/eigen3 -I/usr/include/eigen3/eigen3 -I/usr/include/gazebo-9/gazebo/msgs -I/home/jschloss/firmware/mavlink/include -I/usr/include/Paging -isystem /usr/include/opencv -I/home/jschloss/firmware/Tools/sitl_gazebo/external/OpticalFlow/include -isystem /usr/include/uuid -isystem /usr/include/x86_64-linux-gnu  **-std=c++17 -std=c++11** -Wno-deprecated-declarations -fPIC   -I/usr/include/uuid -I/usr/include/x86_64-linux-gnu **-std=gnu++14** -o CMakeFiles/mav_msgs.dir/CommandMotorSpeed.pb.cc.o -c /home/jschloss/firmware/build/posix_sitl_default/build_gazebo/CommandMotorSpeed.pb.cc

After this patch, only gnu++1z is set.

> /usr/bin/c++  -DLIBBULLET_VERSION=2.87 -DLIBBULLET_VERSION_GT_282 -Dmav_msgs_EXPORTS -isystem /usr/include/gazebo-9 -isystem /usr/include/bullet -isystem /usr/include/simbody -isystem /usr/include/sdformat-6.0 -isystem /usr/include/ignition/math4 -isystem /usr/include/OGRE -isystem /usr/include/OGRE/Terrain -isystem /usr/include/OGRE/Paging -isystem /usr/include/ignition/transport4 -isystem /usr/include/ignition/msgs1 -isystem /usr/include/ignition/common1 -isystem /usr/include/ignition/fuel_tools1 -I/home/jschloss/firmware/Tools/sitl_gazebo/include -I/home/jschloss/firmware/build/posix_sitl_default/build_gazebo -I/usr/include/eigen3 -I/usr/include/eigen3/eigen3 -I/usr/include/gazebo-9/gazebo/msgs -I/home/jschloss/firmware/mavlink/include -I/usr/include/Paging -isystem /usr/include/opencv -I/home/jschloss/firmware/Tools/sitl_gazebo/external/OpticalFlow/include -isystem /usr/include/uuid -isystem /usr/include/x86_64-linux-gnu  -Wno-deprecated-declarations -fPIC   -I/usr/include/uuid -I/usr/include/x86_64-linux-gnu **-std=gnu++1z** -o CMakeFiles/mav_msgs.dir/CommandMotorSpeed.pb.cc.o -c /home/jschloss/firmware/build/posix_sitl_default/build_gazebo/CommandMotorSpeed.pb.cc

### CODE REVIEW

**Review by TSC21 — APPROVED** [review]
Thanks @jacobschloss! Let's wait for CI to pass and then merge.

**Comment by TSC21** [comment]
@jacobschloss this fails for Gazebo7 (Ubuntu 16.04) - https://travis-ci.org/PX4/sitl_gazebo/jobs/452151932

**Comment by jacobschloss** [comment]
Hmm. Looks like support for setting CXX to 17 is spotty before cmake 3.10. This might be hard to do in a backwards-compatible way, short of disabling CXX17 detection on old cmakes (which might be ok, since you would generally have an old compiler too, that does not support C++17.

Edit: no that is bad, clearly the compiler thinks it supports C++17.

**Comment by jacobschloss** [comment]
@TSC21  Would it be better to not try to detect C++17 with old cmake? 98, 11, and 14 are supported at least as far back as CMake 3.1. I can change it to only detect C++17 with cmake at least 3.8.0, and otherwise detect up to c++14?

**Comment by TSC21** [comment]
> I can change it to only detect C++17 with cmake at least 3.8.0, and otherwise detect up to c++14?

Isn't 3.5 enough?

**Comment by jacobschloss** [comment]
https://cmake.org/cmake/help/v3.5/prop_tgt/CXX_STANDARD.html for version 3.5.2 doesn't list 17 as a supported value. I don't have that version to try though.

Oh I think I got my version test backwards. I should test for newer than 3.8.2... one sec

File: cmake/EnableC++XX.cmake [hunk 1]
```diff
@@ -43,4 +43,4 @@ else()
 
 endif()
 
-
+set(CMAKE_CXX_EXTENSIONS OFF)
```
Comment (TSC21) [inline cmake/EnableC++XX.cmake]: This should be conditional as well

**Comment by TSC21** [comment]
Travis CI failure is unrelated with this PR. Merging now. Thanks @jacobschloss!

**Pull request merged**